inference-engine/src/vpu/graph_transformer/src/middleend/passes/propagate_dynamism_to_outputs.cpp

```diff
@@ -39,34 +39,33 @@ class PassImpl final : public Pass {
             }
             const auto parent = parentDataToShapeEdge->parent();
 
+            const auto& parentAttrs = parent->attrs();
+            VPU_THROW_UNLESS(parentAttrs.getOrDefault("converted-notation", false),
+                "All shape parent data object must be already converted to MDK notation");
+
+            const auto& parentProducer = parent->producer();
+            const auto& parentInIENotation = parentProducer->input(0);
+            VPU_THROW_UNLESS(parentInIENotation->usage() == DataUsage::Intermediate,
+                "Shape parent data object is expected to be an intermediate data object since shape child is not an output");
+
+            const auto& parentInIENotationAttrs = parent->attrs();
```

Comment:
@gladilov-gleb, sorry, I have not noticed it before. 
In your code:
`const auto& parentAttrs = parent->attrs();` and `const auto& parentInIENotationAttrs = parent->attrs();` are the same
Probably `const auto& parentInIENotationAttrs = parentInIENotation->attrs();` should be used

Reply:
Nice catch, thanks - fixed